pkg/etcd/resolver.go

```diff
@@ -30,16 +30,29 @@ func (r *EtcdSimpleResolver) Build(target resolver.Target, cc resolver.ClientCon
 		return nil, err
 	}
 
-	if r.CC != nil {
+	if getCC(r) != nil {
 		addr, serverName := interpret(r.endpoint)
+		a := resolver.Address{Addr: addr, ServerName: serverName}
 		r.UpdateState(resolver.State{
-			Addresses: []resolver.Address{{Addr: addr, ServerName: serverName}},
+			Endpoints: []resolver.Endpoint{
+				{Addresses: []resolver.Address{a}},
+			},
 		})
 	}
 
 	return res, nil
 }
 
+func getCC(r *EtcdSimpleResolver) (cc resolver.ClientConn) {
+	defer func() {
+		if rec := recover(); rec != nil {
+			cc = nil
+		}
+	}()
+
+	return r.CC()
```

Comment:
Is this panicking because r is nil, or because `CC()` in `manual.Resolver` panics under some condition? Can we address/prevent calling it if it'll panic, instead of recovering from it?

Reply:
I think I can just get rid of this, because for etcd, they seperate the Build and UpdateState functions, so they have the possibility of updateState being called before a resolver has been created. We don't have that problem. We always call `Resoler.Build` before using the CC

Reply:
> CC returns r's ClientConn when r was last Built. Panics if the resolver has not been Built before.

We're calling this after calling Build, so it shouldn't panic...

Do we need to just change this to:
```suggestion
	return r.Resolver.CC()
```